pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/FilteringObjectInputStream.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ro.pippo.session;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InvalidClassException;
+import java.io.ObjectInputStream;
+import java.io.ObjectStreamClass;
+import java.util.ArrayList;
+
+import static ro.pippo.core.util.StringUtils.isNullOrEmpty;
+
+/**
+ * @author idealzh
+ */
+public class FilteringObjectInputStream extends ObjectInputStream {
+    private ArrayList whiteClassNames = new ArrayList<String>();
+
+    public FilteringObjectInputStream(InputStream in) throws IOException {
+        super(in);
+
+        whiteClassNames.add("ro.pippo.session.DefaultSessionData");

```

Comment:
We can read these information from a file "src/main/resources/pippo/withelist-serialization.txt" where the content of that file is like (one class name on one line - it's easy to read):
```
ro.pippo.session.DefaultSessionData
java.util.HashMap
ro.pippo.core.Flash
java.util.ArrayList
```

For a reference implementation see [this](https://github.com/richfaces4/core/blob/master/impl/src/main/java/org/richfaces/util/LookAheadObjectInputStream.java).